Resources/public/js/components/products/components/attributes/main.js

```diff
@@ -0,0 +1,365 @@
+         */
+        setHeaderInformation = function() {
+            var title = 'pim.product.title',
+                breadcrumb = [
+                    {title: 'navigation.pim'},
+                    {title: 'pim.products.title'}
+                ];
+            if (!!this.options.data && !!this.options.data.name) {
+                title = this.options.data.name;
+            }
+            title = this.sandbox.util.cropTail(title, maxLengthTitle);
+            this.sandbox.emit('sulu.header.set-title', title);
+
+            if (!!this.options.data && !!this.options.data.number) {
+                breadcrumb.push({
+                    title: '#' + this.options.data.number
+                });
+            } else {
+                breadcrumb.push({
+                    title: 'pim.product.title'
+                });
+            }
+            this.sandbox.emit('sulu.header.set-breadcrumb', breadcrumb);
+        },
+
+        /**
+         * Create the overlay
+         */
+        createAddOverlay = function() {
+
+            // call JSON to get the attributes from the server then create the overlay after it's done
+            var jqxhr = $.getJSON("api/attributes", function(data) {
+                this.attrTypes = new Array();
+
+                $.each(data._embedded.attributes, function(key, value) {
+                    var newAttribute = {
+                        "id": value.id,
+                        "name": value.name
+                    };
+
+                    // at this time we support only text type attributes
+                    if (value.type.name === "product.attribute.type.text") {
+                        this.attrTypes.push(newAttribute);
+                    }
+
+                }.bind(this));
+
+            }.bind(this))
+            .done(function() {
+                // create container for overlay
+                var $overlay = this.sandbox.dom.createElement('<div>');
+                this.sandbox.dom.append(this.$el, $overlay);
+
+                // create content
+                this.sandbox.start([
+                    {
+                        name: 'overlay@husky',
+                        options: {
+                            el: $overlay,
+                            supportKeyInput: false,
+                            title: this.sandbox.translate('product.attribute.overlay.title'),
+                            skin: 'normal',
+                            openOnStart: true,
+                            removeOnClose: true,
+                            instanceName: overlayInstanceName,
+                            data: createAddOverlayContent.call(this),
+                            okCallback: overlayOkClicked.bind(this)
+                        }
+                    }
+                ]);
+
+            }.bind(this))
+            .fail(function() {
+                console.log("Error retrieving attributes from server");
+            });
+
+            jqxhr.complete(function() {
```

Comment:
use this syntax for the other events (done, fail) as well.

jqxhr.done()
jqxhr.fail()